### ISSUE TITLE
Emulation: Support geolocation override

### ISSUE THREAD

**Issue by jimevans** [w3c/webdriver-bidi#343]
One request often heard from users is that they want to simulate a browser at a specific geolocation for their sites. WebDriver should provide a mechanism to facilitate this testing.

**Comment by whimboo** [comment]
Thank you @jimevans for filing!

I want to add that clients should be able to set this location override per tab.

**Comment by OrKoN** [comment]
Should we propose it as part of the https://w3c.github.io/geolocation-api/ spec?

**Comment by OrKoN** [comment]
Chrome currently supports latitude, longitude, accuracy overrides which works for the top-level context and iframes. It does not affect the permission and it does not allow to use geolocation on insecure contexts.

**Comment by christian-bromann** [comment]
WebdriverIO currently already supports this by overwriting the `navigator.geolocation` interface in the browser via the preload script:

```ts
const patchedFn = options instanceof Error
    ? `cbError(new Error(${JSON.stringify(options.message)}))`
    : `cbSuccess({
        coords: ${JSON.stringify(options)},
        timestamp: Date.now()
    })`
await this.scriptAddPreloadScript({
    functionDeclaration: /*js*/`() => {
        Object.defineProperty(navigator.geolocation, 'getCurrentPosition', {
            value: (cbSuccess, cbError) => ${patchedFn}
        })
    }`
})
```

**Comment by OrKoN** [comment]
@christian-bromann Thanks for sharing! Do you support changing it after the page is loaded/during the test? Also, what about `Geolocation.watchPosition()`?

I think the override on the browser side would be more robust but preload scripts offer the ability to implement something already.

**Comment by christian-bromann** [comment]
I agree. The current emulation features we have can't be changed after the page has loaded. It would be helpful, especially for component tests to modify these capabilities without having to do a page load.

**Comment by whimboo** [comment]
Note that there is an open PR for the geolocation spec for WebDriver integration: https://github.com/w3c/geolocation/pull/170

**Comment by whimboo** [comment]
And [here are the minutes](https://docs.google.com/document/d/1EtEFfQ3sJ6lECF3ojmbLKIH7iKN4RdGrapVosQB-wqI/edit#heading=h.hzkstbqqa0cj) from yesterday's joined meeting with the WebApplication group.
